components/script/dom/element.rs

```diff
@@ -2576,6 +2605,19 @@ impl Element {
         self.set_click_in_progress(false);
     }
 
+    // https://html.spec.whatwg.org/multipage/#language
+    pub fn get_lang(&self) -> String {
+        self.upcast::<Node>().inclusive_ancestors().filter_map(|node| {
+            node.downcast::<Element>().and_then(|el| {
+                el.get_attribute(&ns!(xml), &local_name!("lang")).map_or_else(|| {
```

Comment:
Let's use here `or_else` and then just one `map` to avoid the `String::from(attr.Value())` duplication.